src/core/content/connector.js

```diff
@@ -334,6 +355,10 @@ function BaseConnector() {
 			newState.track = artistTrack.track;
 		}
 
+		if (newState.artist === newState.albumArtist) {
```

Comment:
@alexesprit this is before the filter is applied? where would be a good place to hook up this sort of validation or do you think this is alright to push both fields when they are the same?

Reply:
Yes, in this block you compare raw data (i.e. from the page).

I think there could be cases when raw artist and raw albumArtist are different, mainly because of leading/trailing spaces. Probably, it's much better to compape right before sending the **filtered** connector state to the controller.

It could be fixed by trimming directly in the getters, but I think it's a dirty hack, and I would not recommend to go this way if there's another solution.

> or do you think this is alright to push both fields when they are the same

I don't know how Last.fm responses if we post both artist and albumArtist if they're the same. We can just test it and see how things are going. If it's okay, then this check is redundant.

Reply:
Dirty hacks I'm not a fan of. Added a pre-process function to act on the filtered state.

Reply:
Yes, that's better, althougth I'd move that block to `Connector.filterState` function instead of making the new one. I consider removing delete.albumArtist is a part of filtering.